src/main.cpp

```diff
@@ -22,6 +22,44 @@
 #include "application.h"
 #endif
 
+#ifndef Q_WS_WIN
+#include <signal.h>
+#endif
+
+namespace {
+
+#ifndef Q_WS_WIN // windows don't need these handlers
+
+void sigintHandler(int /* unused*/)
+{
+    qDebug("[signal] SIGINT received, exiting");
+    QCoreApplication::exit();
+}
+
+bool setSignalHandler()
+{
+    struct sigaction sigint_action;
+    sigint_action.sa_handler = sigintHandler;
+    sigemptyset(&sigint_action.sa_mask);
+    sigint_action.sa_flags = 0;
+    sigint_action.sa_flags |= SA_RESTART;
+    if (sigaction(SIGINT, &sigint_action, 0) != 0) // 0 if okay, -1 if failed
+        return false;
+    return true;
```

Comment:
In fact there is a document from qt for it : https://qt-project.org/doc/qt-4.8/unix-signals.html